### PR TITLE
Bugfixes for cases in which important points are not in the final geometry

### PR DESCRIPTION
**Pull request by fhagemann**
This PR fixes the errors described in #256.

### CODE REVIEW

**Review by hervasa2 — APPROVED** [review]
Looks good! Thank you

**Comment by fhagemann** [comment]
The only thing is that the warning appears twice when creating this plot
(apparently `get_scale` is called twice):
samplesurface

Is this intended? @hervasa2

**Comment by hervasa2** [comment]
> The only thing is that the warning appears twice when creating this plot (apparently `get_scale` is called twice):
> 
> ```julia
> using SolidStateDetectors
> using Plots; gr(fmt = :png)
> sim = Simulation("MWEsamplesurface.yaml");
> plot(sim.detector.semiconductor, size = (500,500), st = :mesh3d, color = :lightgray, lw = 0.5)
> plot(sim.detector, show_semiconductor = true, st = :samplesurface, size = (500,500), n_samples = 50)
> ```
> 
> samplesurface
> 
> Is this intended? @hervasa2

`get_scale` gets called for each csg object so this is expected when plotting contacts.

**Comment by fhagemann** [comment]
> `get_scale` gets called for each csg object so this is expected when plotting contacts.

The thing is that the only CSG object here is the semiconductor. If I only plot the contacts (both are just simple tubes, no CSG), I don’t get the warning at all.

**Comment by lmh91** [comment]
> The thing is that the only CSG object here is the semiconductor. If I only plot the contacts (both are just simple tubes, no CSG), I don’t get the warning at all.
> 
> It also looks like I'm plotting more points for the contacts by just "removing" the semiconductor:

In the plot recipe where the `get_scaled` is actually called one would need to get the scales for all objects, not just the ones which are to be plotted. I think it would be okay to do so.

File: src/ConstructiveSolidGeometry/plotting/CSG/CSG.jl [hunk 1]
```diff
@@ -6,10 +6,12 @@ function primitives(csg::AbstractConstructiveGeometry)
     ps
 end
 
-@recipe function f(csg::AbstractConstructiveGeometry{T}; n_samples = 40) where {T}
+@recipe function f(csg::AbstractConstructiveGeometry{T}; n_samples = 40, CSG_scale = missing) where {T}
     seriestype --> :csg
     ps = primitives(csg)
-    spacing = (haskey(plotattributes, :seriestype) && plotattributes[:seriestype] == :samplesurface) ?  T(get_scale(csg)/n_samples) : nothing
+    if haskey(plotattributes, :seriestype) && plotattributes[:seriestype] == :samplesurface
+        spacing::T = T((ismissing(CSG_scale) ? get_scale(csg) : CSG_scale)/n_samples)
```
Comment (fhagemann) [inline src/ConstructiveSolidGeometry/plotting/CSG/CSG.jl:13]: Now, the scales are only determined once for a detector and passed to the plot recipes of the individual geometries.
However, if the geometries/objects are plotted by themselves, the scales are determined within that

**Review by fhagemann — COMMENTED** [review]
The scales are now only determined once when plotting.

Examples:
```julia
plot(sim.detector, show_semiconductor = true, n_samples = 50, st = :samplesurface, ratio = 1, size = (500,500))
```
```
┌ Warning: The number of samples might be smaller than expected. Consider increasing `n_samples` to obtain the desired result.
└ @ SolidStateDetectors.ConstructiveSolidGeometry /home/mpp/Software/SolidStateDetectors.jl/src/ConstructiveSolidGeometry/CSG.jl:223
```
detector

```julia
plot(sim.detector.semiconductor, n_samples = 50, st = :samplesurface, ratio = 1, size = (500,500))
```
```
┌ Warning: The number of samples might be smaller than expected. Consider increasing `n_samples` to obtain the desired result.
└ @ SolidStateDetectors.ConstructiveSolidGeometry /home/mpp/Software/SolidStateDetectors.jl/src/ConstructiveSolidGeometry/CSG.jl:223
```
semiconductor

```julia
plot(sim.detector, n_samples = 50, st = :samplesurface, ratio = 1, size = (500,500))
```
contacts

**Pull request merged**